lib/ts/recipe/dev/recipeImplementation.ts

```diff
@@ -0,0 +1,70 @@
+import { RecipeInterface } from "./types";
+import { Querier } from "../../querier";
+import NormalisedURLPath from "../../normalisedURLPath";
+import { isUsingDevelopmentClientId } from "./utils";
+import { USING_DEV_CORE, USING_DEV_OAUTH_KEY, INVALID_API_KEY, NO_API_KEY } from "./constants";
+
+export default class RecipeImplementation implements RecipeInterface {
+    querier: Querier;
+    constructor(querier: Querier) {
+        this.querier = querier;
+    }
+
+    checkConnectionToCore = async (
```

Comment:
Terrible name.. this does more than just check the connection.. It also checks the third party keys.